browser/net/url_context.h

```diff
@@ -102,7 +102,9 @@ struct BraveRequestInfo {
   GURL ipfs_gateway_url;
   bool ipfs_auto_fallback = false;
 
-  bool ShouldMockRequest() const { return !mock_data_url.empty(); }
+  bool ShouldMockRequest() const {
+    return blocked_by != kNotBlocked && !mock_data_url.empty();
+  }
```

Comment:
@iefremov one thing I had to change in `adblock-rust` is that the presence of a redirect no longer implies blocking - but browser side we only want to use the redirection if the request is blocked.